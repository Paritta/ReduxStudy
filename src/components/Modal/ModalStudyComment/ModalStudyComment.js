import React from "react";
import FaUser from "react-icons/lib/fa/user";
import PropTypes from "prop-types";
import { commentReceiveRequest } from "../../../modules/CommentReceive";
import { connect } from "react-redux";
import { getCommentReceive } from "../../../selector";
import styled from "styled-components";
import oc from "open-color";

const propTypes = {
    CommentId: PropTypes.string.isRequired,
    CommentReceive: PropTypes.object.isRequired,
};

const defaultTypes = {
    CommentId: "",
    CommentReceive: {}
};

const Wrapper = styled.div`
    height: 10%;
    width: 100%;
    
    display: flex;
    flex-direction: row;
    
    position: relative;
    
    padding: 10px 15px;
`;

const CommentWrapper = styled.div`
    display: flex;
    flex-direction: column;
    
    padding-left: 10px;
`;

const Comment = styled.div`
    font-size: 15px;
    font-weight: 300;
`;

const NickName = styled.div`
    font-size: 15px;
    font-weight: 300;
`;

const CommentIcon = styled.div`
    height: 35px;
    width: 35px;
   
    border-radius: 100%;
    
    background: ${oc.gray[2]};
    
    top: 10px;
    left: 20px;
    
    display: flex;
    justify-content: center;
    align-items: center;
`;

const Line = styled.div`
    background: black;
    height: 1px;
`;

export class ModalStudyComment extends React.Component {
    render () {
        const { CommentReceive } = this.props;

        return (
            <div>
            <Wrapper>
                <CommentIcon>
                    <FaUser size={25} color="white"/>
                </CommentIcon>
                <CommentWrapper>
                    {
                        !CommentReceive.pending && CommentReceive.data.length !== 0 &&
                        <Comment>
                            { CommentReceive.data[this.props.CommentId].comment }
                        </Comment>
                    }
                    {
                        !CommentReceive.pending && CommentReceive.data.length !== 0 &&
                        <NickName>
                            { CommentReceive.data[this.props.CommentId].NickName }
                        </NickName>
                    }
                </CommentWrapper>
            </Wrapper>
            <Line />
            </div>
        )
    }
}

ModalStudyComment.propTypes = propTypes;
ModalStudyComment.defaultTypes = defaultTypes;

export default connect(
    state => ({ CommentReceive: getCommentReceive(state) }),
    { commentReceiveRequest }
)(ModalStudyComment);

