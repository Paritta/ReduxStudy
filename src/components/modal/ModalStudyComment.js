import React from "react";
import PropTypes from "prop-types";

const propTypes = {
};

const defaultTypes = {
};

export const ModalStudyComment = () => {
    return (
        <div>
        </div>
    )
};

ModalStudyComment.propTypes = propTypes;
ModalStudyComment.defaultTypes = defaultTypes;

export default ModalStudyComment;
