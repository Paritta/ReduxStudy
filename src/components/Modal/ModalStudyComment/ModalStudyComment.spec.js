import React from "react";
import { ModalStudyComment } from "./ModalStudyComment";
import "jest-styled-components";
import ReactTestRenderer from "react-test-renderer";

it("Layout render correctly", () => {
    const CommentReceive = {
        pending: false,
        data: []
    };

    const tree = ReactTestRenderer.create(
        <ModalStudyComment
            CommentReceive={CommentReceive}
        />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
