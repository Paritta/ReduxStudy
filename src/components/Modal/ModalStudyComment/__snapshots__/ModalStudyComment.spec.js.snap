// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render correctly 1`] = `
.c0 {
  height: 10%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 10px 15px;
}

.c1 {
  height: 35px;
  width: 35px;
  border-radius: 100%;
  background: #e9ecef;
  top: 10px;
  left: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 10px;
}

.c3 {
  background: black;
  height: 1px;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <svg
        fill="currentColor"
        height={25}
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": "white",
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width={25}
      >
        <g>
          <path
            d="m35.9 31.4q0 2.6-1.6 4.2t-4.3 1.5h-19.5q-2.7 0-4.4-1.5t-1.6-4.2q0-1.2 0.1-2.3t0.3-2.5 0.6-2.4 0.9-2.2 1.4-1.8 1.9-1.2 2.5-0.4q0.2 0 1 0.5t1.6 1 2.4 1.1 3 0.5 3-0.5 2.4-1.1 1.7-1 0.9-0.5q1.4 0 2.5 0.4t1.9 1.2 1.4 1.8 0.9 2.2 0.6 2.4 0.4 2.5 0 2.3z m-7.1-20q0 3.6-2.5 6.1t-6.1 2.5-6-2.5-2.6-6.1 2.6-6 6-2.5 6.1 2.5 2.5 6z"
          />
        </g>
      </svg>
    </div>
    <div
      className="c2"
    />
  </div>
  <div
    className="c3"
  />
</div>
`;
